modules/xwayland/xwayland.yml

```diff
@@ -11,9 +11,25 @@ config-opts:
 sources:
 - type: git
   url: https://gitlab.freedesktop.org/xorg/xserver.git
-  tag: xwayland-21.1.2
-  commit: 1dbb96ae48f1fa69752e71f25f03d2f733918ffb
+  tag: xwayland-21.1.4
+  commit: fbc03d7e486e8c1e33f0c2d3ff9ed415c95de166
 modules:
+- name: wayland
+  buildsystem: meson
+  sources:
+  - type: git
+    url: https://github.com/wayland-project/wayland
+    tag: 1.20.0
```

Comment:
Freedesktop-sdk 21.08 contains wayland 1.20.0, why bundle it?

Reply:
There may be an issue with my flatpak setup? But in my system (with sdk 21.08) flatpak builder refused to build wlroots because wayland was 1.19.0.

Reply:
Maybe it was updated. Try again, dropping the wayland module (and probably libdrm module, too).

Reply:
I just checked, race condition: 21.08.10 came out in between with wayland 1.20.0, 21.08.9 had wayland 1.19.0.

Reply:
This package has been unable to update for the **past two months.**

Upstream gamescope updated to wlroots 0.15.0 on the 14th of December 2021, last year.
wlroots 0.15.0 requires Wayland 1.20 because they have a new Vulkan renderer that **needs** Wayland 1.20

gamescope now exclusively uses this new Vulkan renderer after planing on dropping all legacy OpenGL and EGL code for a while [#202](https://github.com/Plagman/gamescope/pull/202)
All of this work accumulated in the 3.10.0 gamescope release on the 14th of December 2021, last year.

Since 3.10.0, there has been **15** further stable releases:

- 3.10.0 - 3.10.8
- 3.11.0 - 3.11.5

All of which **needs** Wayland 1.20 or higher to work, but the freedesktop-sdk-21.08 stayed on Wayland 1.19 until finally updating with the 21.08.10 release, [on the 1st of February 2022](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/tags?search=21.08&sort=updated_desc)

Any attempt to try and update this package before now resulted in a failed build.
Just look at my failed closed pull requests:

```
wlroots| Project name: wlroots
wlroots| Project version: 0.15.0
wlroots| C compiler for the host machine: ccache cc (gcc 11.2.0 "cc (GCC) 11.2.0")
wlroots| C linker for the host machine: cc ld.bfd 2.37
wlroots| Compiler for C supports arguments -Wundef: YES
wlroots| Compiler for C supports arguments -Wlogical-op: YES
wlroots| Compiler for C supports arguments -Wmissing-include-dirs: YES
wlroots| Compiler for C supports arguments -Wold-style-definition: YES
wlroots| Compiler for C supports arguments -Wpointer-arith: YES
wlroots| Compiler for C supports arguments -Winit-self: YES
wlroots| Compiler for C supports arguments -Wstrict-prototypes: YES
wlroots| Compiler for C supports arguments -Wimplicit-fallthrough=2: YES
wlroots| Compiler for C supports arguments -Wendif-labels: YES
wlroots| Compiler for C supports arguments -Wstrict-aliasing=2: YES
wlroots| Compiler for C supports arguments -Woverflow: YES
wlroots| Compiler for C supports arguments -Wmissing-prototypes: YES
wlroots| Compiler for C supports arguments -Walloca: YES
wlroots| Compiler for C supports arguments -Wno-missing-braces: YES
wlroots| Compiler for C supports arguments -Wno-missing-field-initializers: YES
wlroots| Compiler for C supports arguments -Wno-unused-parameter: YES
wlroots| Compiler for C supports arguments -fmacro-prefix-map=/prefix/to/hide=: YES
wlroots| Dependency wayland-server found: NO found 1.19.0 but need: '>=1.20' (cached)

../subprojects/wlroots/meson.build:101:0: ERROR: Dependency 'wayland-server' is required but not found.

A full log can be found at /run/build/gamescope/_flatpak_build/meson-logs/meson-log.txt
FB: host_command_exited_cb 2124704 256

�]2;flatpak-builder: Cleanup gamescope�Error: module gamescope: Child process exited with code 1
FB: Unmounting read-only fs: fusermount -uz /srv/buildbot/worker/build-x86_64-2/build/.flatpak-builder/rofiles/rofiles-xo5qVn
dbus-daemon[2096392]: Reloaded configuration
```

It was just a coincidence that @ganguin managed to fix the build at around the same time freedesktop-sdk finally updated their Wayland version.
To prevent such long delays in the future, I think we should continue to bundle our own version of Wayland.

The only issue now is that when we finally have a working build, I can't raise @xinyazhang to push out the update.
I've been pinging him to no response :(

Reply:
>To prevent such long delays in the future, I think we should continue to bundle our own version of Wayland.

We can always bundle an updated dep again if it proves necessary. But as of now freedesktop-sdk includes sufficiently new wayland, so bundling it just makes the flatpak build more complex for no good reason.

Reply:
I will take heed of your experience in this matter @gasinvein to drop the bundling of Wayland.

But I will be saving a copy of @ganguin's work here (and watching some more YouTube tutorials), so that in the event that we **need** to rebundle I will have a better chance of pulling it off.